src/makeCalculator.js

```diff
@@ -4,7 +4,49 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  let result = 0;
+
+  return {
+    get result() {
+      return result;
+    },
+
+    add: (num) => {
+      result += num;
+
+      return this;
+    },
+
+    subtract: (num) => {
+      result -= num;
+
+      return this;
+    },
+
+    multiply: (num) => {
+      result *= num;
+
+      return this;
+    },
+
+    divide: (num) => {
+      result /= num;
+
+      return this;
+    },
+
+    reset() {
+      result = 0;
+
+      return this;
+    },
+
+    operate(operation, num) {
+      operation.call(this, num);
+
+      return this;
```

Comment:
[CODE KNOWLEDGE] - Since you are using the 'operate' method to call the passed operation, there is no need to use 'operation.call(this, num)'. You can directly invoke the operation as 'operation(num)' because 'this' binding is not required here.